### PR TITLE
Added pivotConfig option to alias series with a prefix

### PR DESCRIPTION
**Pull request by MattGson**
Allows for multiple series using the same measure in
blending queries.

Currently works in seriesNames and chartPivot only

**Check List**
- [☑️ ] Tests has been run in packages where changes made if available
- [☑️ ] Linter has been run for changed code
- [☑️ ] Tests for the changes have been added if not covered yet
- [☑️ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

#1573

**Description of Changes Made (if issue reference is not provided)**

Added an option to `pivotConfig` called `aliasSeries ` with type `string[]`.
This takes a list of series aliases which are prefixed to the corresponding series in the pivot.

This has only been implemented for the `chartPivot` and `seriesNames` methods currently.

**Note**: There seems to be a lot of line changes due to the existing code not following the `prettier` guidelines.
Actual changes at 435 and 592

### CODE REVIEW

File: packages/cubejs-client-core/index.d.ts [hunk 1]
```diff
@@ -178,6 +178,10 @@ declare module '@cubejs-client/core' {
      * If `true` missing dates on the time dimensions will be filled with `0` for all measures.Note: the `fillMissingDates` option set to `true` will override any **order** applied to the query
      */
     fillMissingDates?: boolean | null;
+    /**
+     * Give each series a prefix alias. Should have one entry for each query:measure.
+     */
+    aliasSeries: string[];
```
Comment (vasilev-alex) [inline packages/cubejs-client-core/index.d.ts:184]: This should probably be optional

**Review by vasilev-alex — CHANGES_REQUESTED** [review]
Hey! Thanks for your time submitting this PR.

I think we could even make the aliasing thing automatic when no `pivotConfig` is provided, so it doesn't come as a surprise resulting in missing all duplicate measures. What do you think?

Also, could you please only submit the actual changes without any formatting applied by prettier as it currently breaks our linter rules and makes history difficult to follow.

File: packages/cubejs-client-core/index.d.ts (hunk 1 above)
Comment (MattGson) [reply packages/cubejs-client-core/index.d.ts]: Fixed

**Comment by MattGson** [comment]
@vasilev-alex I have made the suggested changes including an automatic alias for duplicate measures.

Currently the alias will just use the index i.e. 0,Users.Count 1,Users.count ...

There is potential for some abstractions/re-use around the aliasing code but I'm not super familiar with the code-base and the ResultSet code isn't the easiest to follow (Typescript would be a big improvement!).

**Pull request merged**